rxandroidble/src/main/java/com/polidea/rxandroidble/internal/scan/ExcessiveScanCheckerApi24.java

```diff
@@ -0,0 +1,45 @@
+package com.polidea.rxandroidble.internal.scan;
+
+
+import android.support.annotation.Nullable;
+import java.util.Date;
+
+public class ExcessiveScanCheckerApi24 implements ExcessiveScanChecker {
+
+    /*
+     * default values taken from
+     * https://android.googlesource.com/platform/packages/apps/Bluetooth/+/android-7.0.0_r1/src/com/android/bluetooth/gatt/AppScanStats.java
+     */
+    private static final int SCANS_LENGTH = 5;
+
+    private static final long EXCESSIVE_SCANNING_PERIOD = 30 * 1000;
+
+    private final long[] previousChecks = new long[SCANS_LENGTH];
+
+    @Nullable
+    @Override
+    public Date suggestDateToRetry() {
+        // TODO: [DS] 27.06.2017 Think if persisting this information through Application close is needed
+        final int oldestCheckTimestampIndex = getOldestCheckTimestampIndex();
+        final long oldestCheckTimestamp = previousChecks[oldestCheckTimestampIndex];
+        final long currentCheckTimestamp = System.currentTimeMillis();
+        if (currentCheckTimestamp - oldestCheckTimestamp < EXCESSIVE_SCANNING_PERIOD) {
+            return new Date(oldestCheckTimestamp + EXCESSIVE_SCANNING_PERIOD);
+        }
+        previousChecks[oldestCheckTimestampIndex] = currentCheckTimestamp;
+        return null;
+    }
+
+    private int getOldestCheckTimestampIndex() {
```

Comment:
Something like a fixed size stack seems to be cool https://stackoverflow.com/a/16206356